ocs_ci/helpers/managed_services.py

```diff
@@ -453,14 +462,15 @@ def verify_faas_provider_resources():
     )
 
     # Verify that Cephcluster is Ready and hostNetworking is True
-    cephcluster = OCP(
-        kind=constants.CEPH_CLUSTER, namespace=config.ENV_DATA["cluster_namespace"]
+    cephcluster = ResourceWithPhase(
+        kind=constants.CEPH_CLUSTER,
+        namespace=config.ENV_DATA["cluster_namespace"],
+        resource_name=constants.CEPH_CLUSTER_NAME,
     )
-    cephcluster_yaml = cephcluster.get().get("items")[0]
-    log.info("Verifying that Cephcluster is Ready and hostNetworking is True")
-    assert (
-        cephcluster_yaml["status"]["phase"] == "Ready"
-    ), f"Status of cephcluster {cephcluster_yaml['metadata']['name']} is {cephcluster_yaml['status']['phase']}"
+    log.info("Waiting for Cephcluster to be Ready")
```

Comment:
This is implemented differently in the function 'verify_provider_resources ' in this PR. Isn't that suitable here ?

Reply:
I think using wait_for_phase function is preferable to re-implementing it using TimeoutSampler.